dali/kernels/signal/resampling.h

```diff
@@ -49,6 +52,22 @@ struct ResamplingWindow {
     return lookup[i] + di * (lookup[i + 1] - lookup[i]);
   }
 
+#ifdef __SSE2__
+  inline __m128 operator()(__m128 x) const {
+    __m128 fi = _mm_add_ps(x * _mm_set1_ps(scale), _mm_set1_ps(center));
```

Comment:
Don't wee need to assure that `idx[3]+1 < lookup.size()`?

Reply:
It's externally assured, just as it was for scalar variant that `i+1 < lookup.size()`. Or do you mean to add assert?

Reply:
I'm rather asking if we need such assert. If not then I'm fine too.

Reply:
The assert was there in the scalar variant for debugging. I don't remember if it ever fired, but I kept it there. I'll probably put one here too, if we decide to extend SIMD support to multi-channel audio (now only 1ch variant got the upgrade). I think that for most scenarios it's OK, but there certainly are some where multi-channel audio is required.